CatsAndDogs/ContentLoader.swift

```diff
@@ -16,11 +16,18 @@ private extension Data {
 }
 
 class OnlineContentLoader {
-    let url = URL(string: "https://catsanddogs-kotlin-bff.herokuapp.com/schedule.json?from=2017-05-08T12:00:00-02:00")
+    let urlString = "https://catsanddogs-kotlin-bff.herokuapp.com/kotlinconf/schedule.json"
     let session = URLSession(configuration: .default)
 
     private func downloadSchedule(completion: @escaping (Data?)->()) {
-        session.dataTask(with: url!) { (data, _, _) in completion(data) }.resume()
+        if let url = URL(string: urlString) {
```

Comment:
any better way to do that? In kotlin I would do:
```kotlin
URL(string: urlString)?.let { url ->
   var request = URLRequest(url)
   Bundle.main.preferredLocalizations.first?.let { language ->
      request.setValue("Accept-Language", forHTTPHeaderField = language)
   }
}

Reply:
also failable initialisers are interesting, I'm not sure if I like or not but they don't exist in kotlin. An initialiser cannot return an optional.

Reply:
and a last thing, I tried to write something like that but it couldn't compile (also works in kotlin):
```swift
let request = if let url = URL(string: urlString) { URLRequest(url: url) }
```

Reply:
I'll push up a way to do the URL which is more like your URL.

Reply:
Failable initialisers are probably a necessary evil because of interactions with ObjC where it was commonplace for an init to fail and return nil. I think I like them anyway, but I am not certain about that - they do end up making the code messier (lots of optional chaining/unwrapping) but then they also state clearly that something bad happened (you tried to create an object with a bad set of parameters and so nothing has been created.